cocos/gfx/webgl/webgl-commands.ts

```diff
@@ -2714,10 +2715,30 @@ export function WebGLCmdFuncCopyTexImagesToTexture (
     case gl.TEXTURE_2D: {
         for (let i = 0; i < regions.length; i++) {
             const region = regions[i];
-            // console.debug('Copying image to texture 2D: ' + region.texExtent.width + ' x ' + region.texExtent.height);
-            gl.texSubImage2D(gl.TEXTURE_2D, region.texSubres.mipLevel,
-                region.texOffset.x, region.texOffset.y,
-                gpuTexture.glFormat, gpuTexture.glType, texImages[n++]);
+            if (WECHAT_MINI_PROGRAM) {
+                // TODO: Property 'type' does not exist on type 'HTMLImageElement'.
+                // maybe we need a higher level implementation called `pal/image`, we provide `type` interface here.
+                // issue: https://github.com/cocos/cocos-engine/issues/14646
+                const texImg = texImages[n++] as any;
+                if (texImg.type !== 'undefined' && texImg.type === 'canvas') {
```

Comment:
It is better to hack in platform layer.

Reply:
We do not have this platform layer yet. Similar treatment is here: https://github.com/cocos/cocos-engine/pull/14638/commits/c268b8982cb9975e95b5f1ef855a681eff82b059

Reply:
Can hack in upper layer, not in gfx?

Reply:
move to PAL